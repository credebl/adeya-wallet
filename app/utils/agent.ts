import { Agent, getAgentModules, useAdeyaAgent } from '@adeya/ssi'

export type AdeyaAgent = Agent<ReturnType<typeof getAgentModules>>
export type AdeyaAgentModules = ReturnType<typeof getAgentModules>

interface MyAgentContextInterface {
  loading: boolean
  agent: AdeyaAgent
  setAgent: (agent: AdeyaAgent) => void
}

export const useAppAgent = useAdeyaAgent as () => MyAgentContextInterface
