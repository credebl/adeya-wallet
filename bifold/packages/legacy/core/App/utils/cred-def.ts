import { CredentialMetadataKeys, CredentialExchangeRecord as CredentialRecord } from '@aries-framework/core'

import { parseSchemaFromId, credentialSchema } from './schema'

export function parseCredDefFromId(credDefId?: string, schemaId?: string): string {
  let name = 'Credential'
  if (credDefId) {
    const credDefRegex = /[^:]+/g
    const credDefIdParts = credDefId.match(credDefRegex)
    if (credDefIdParts?.length === 5) {
      name = `${credDefIdParts?.[4].replace(/_|-/g, ' ')}`
        .split(' ')
        .map((credIdPart) => credIdPart.charAt(0).toUpperCase() + credIdPart.substring(1))
        .join(' ')
    }
  }
  if (name.toLocaleLowerCase() === 'default' || name.toLowerCase() === 'credential') {
    name = parseSchemaFromId(schemaId).name
  }
  return name
}

function credentialDefinition(credential: CredentialRecord): string | undefined {
  return credential.metadata.get(CredentialMetadataKeys.IndyCredential)?.credentialDefinitionId
}

export function parsedCredDefNameFromCredential(credential: CredentialRecord): string {
  return parseCredDefFromId(credentialDefinition(credential), credentialSchema(credential))
}

export function parsedCredDefName(credentialDefinitionId: string, schemaId: string): string {
  return parseCredDefFromId(credentialDefinitionId, schemaId)
}
