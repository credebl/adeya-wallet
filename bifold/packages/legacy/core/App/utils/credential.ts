import { CredentialExchangeRecord, CredentialMetadataKeys, CredentialState } from '@aries-framework/core'
import { ImageSourcePropType } from 'react-native'

import { luminanceForHexColor } from './luminance'

export const isValidIndyCredential = (credential: CredentialExchangeRecord) => {
  return (
    credential &&
    (credential.state === CredentialState.OfferReceived ||
      credential.credentials.find((c) => c.credentialRecordType === 'indy'))
  )
}

// eslint-disable-next-line @typescript-eslint/no-explicit-any
export const credentialTextColor = (ColorPallet: any, hex?: string) => {
  const midpoint = 255 / 2
  if ((luminanceForHexColor(hex ?? '') ?? 0) >= midpoint) {
    return ColorPallet.grayscale.darkGrey
  }
  return ColorPallet.grayscale.white
}

export const toImageSource = (source: unknown): ImageSourcePropType => {
  if (typeof source === 'string') {
    return { uri: source as string }
  }
  return source as ImageSourcePropType
}

export const getCredentialIdentifiers = (credential: CredentialExchangeRecord) => {
  return {
    credentialDefinitionId: credential.metadata.get(CredentialMetadataKeys.IndyCredential)?.credentialDefinitionId,
    schemaId: credential.metadata.get(CredentialMetadataKeys.IndyCredential)?.schemaId,
  }
}
