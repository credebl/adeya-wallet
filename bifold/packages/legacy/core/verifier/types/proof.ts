import { ProofIdentifier } from '@aries-framework/core'

export interface MissingAttribute {
  name: string
}

export interface SharedAttribute {
  name: string
  value: string
  identifiers: ProofIdentifier
}

export interface SharedGroupedAttribute {
  name: string
  value: string
}

export interface SharedAttributesGroup {
  attributes: Array<SharedGroupedAttribute>
  identifiers: ProofIdentifier
}

export interface ResolvedPredicate {
  name: string
  predicateType: string
  predicateValue: number
  identifiers: ProofIdentifier
}

export interface UnresolvedPredicate {
  name: string
  predicateType: string
  predicateValue: number
}

export class ParsedIndyProof {
  public sharedAttributes: Array<SharedAttribute>
  public sharedAttributeGroups: Array<SharedAttributesGroup>
  public resolvedPredicates: Array<ResolvedPredicate>
  public unresolvedAttributes: Array<MissingAttribute>
  public unresolvedAttributeGroups: Array<Array<MissingAttribute>>
  public unresolvedPredicates: Array<UnresolvedPredicate>

  public constructor() {
    this.sharedAttributes = []
    this.sharedAttributeGroups = []
    this.resolvedPredicates = []
    this.unresolvedAttributes = []
    this.unresolvedAttributeGroups = []
    this.unresolvedPredicates = []
  }
}

export class CredentialSharedProofData {
  public sharedAttributes: Array<SharedAttribute>
  public sharedAttributeGroups: Array<SharedAttributesGroup>
  public resolvedPredicates: Array<ResolvedPredicate>

  public constructor() {
    this.sharedAttributes = []
    this.sharedAttributeGroups = []
    this.resolvedPredicates = []
  }
}

export type GroupedSharedProofDataItem = { data: CredentialSharedProofData; identifiers: ProofIdentifier }
export type GroupedSharedProofData = Map<string, GroupedSharedProofDataItem>
