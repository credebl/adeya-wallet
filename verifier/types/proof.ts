import { AnonCredsProof } from '@aries-framework/anoncreds'
type AnonCredsProofIdentifier = AnonCredsProof['identifiers'][number]

export interface MissingAttribute {
  name: string
}

export interface SharedAttribute {
  name: string
  value: string
  identifiers: AnonCredsProofIdentifier
}

export interface SharedGroupedAttribute {
  name: string
  value: string
}

export interface SharedAttributesGroup {
  attributes: Array<SharedGroupedAttribute>
  identifiers: AnonCredsProofIdentifier
}

export interface ResolvedPredicate {
  name: string
  predicateType: string
  predicateValue: number
  identifiers: AnonCredsProofIdentifier
}

export interface UnresolvedPredicate {
  name: string
  predicateType: string
  predicateValue: number
}

export class ParsedAnonCredsProof {
  public sharedAttributes: Array<SharedAttribute>
  public sharedAttributeGroups: Array<SharedAttributesGroup>
  public resolvedPredicates: Array<ResolvedPredicate>
  public unresolvedAttributes: Array<MissingAttribute>
  public unresolvedAttributeGroups: Array<Array<MissingAttribute>>
  public unresolvedPredicates: Array<UnresolvedPredicate>

  public constructor() {
    this.sharedAttributes = []
    this.sharedAttributeGroups = []
    this.resolvedPredicates = []
    this.unresolvedAttributes = []
    this.unresolvedAttributeGroups = []
    this.unresolvedPredicates = []
  }
}

export class CredentialSharedProofData {
  public sharedAttributes: Array<SharedAttribute>
  public sharedAttributeGroups: Array<SharedAttributesGroup>
  public resolvedPredicates: Array<ResolvedPredicate>

  public constructor() {
    this.sharedAttributes = []
    this.sharedAttributeGroups = []
    this.resolvedPredicates = []
  }
}

export type GroupedSharedProofDataItem = { data: CredentialSharedProofData; identifiers: AnonCredsProofIdentifier }
export type GroupedSharedProofData = Map<string, GroupedSharedProofDataItem>
