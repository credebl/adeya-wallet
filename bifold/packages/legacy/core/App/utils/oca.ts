import { CredentialExchangeRecord } from '@aries-framework/core'

import { CredentialSharedProofData, IndyProofRequestTemplatePayloadData } from '../../verifier'
import { Attribute, Field, Predicate } from '../types/record'

export const buildFieldsFromIndyCredential = (credential: CredentialExchangeRecord): Array<Field> => {
  return credential?.credentialAttributes?.map((attr) => new Attribute(attr)) || []
}

export const buildFieldsFromIndyProofRequestTemplate = (data: IndyProofRequestTemplatePayloadData): Array<Field> => {
  const fields = []
  if (data.requestedAttributes) {
    for (const item of data.requestedAttributes) {
      if (item.name) {
        fields.push(new Attribute({ name: item.name, value: null, ...item }))
      }
      if (item.names) {
        for (const name of item.names) {
          fields.push(new Attribute({ name, value: null, ...item }))
        }
      }
    }
  }
  if (data.requestedPredicates) {
    for (const item of data.requestedPredicates) {
      fields.push(
        new Predicate({
          pType: item.predicateType,
          pValue: item.predicateValue,
          ...item,
        })
      )
    }
  }
  return fields
}

export const buildFieldsFromSharedIndyProof = (data: CredentialSharedProofData): Array<Field> => {
  const fields = []
  for (const attribute of data.sharedAttributes) {
    fields.push(
      new Attribute({
        name: attribute.name,
        value: attribute.value,
      })
    )
  }
  for (const attributesGroup of data.sharedAttributeGroups) {
    for (const attribute of attributesGroup.attributes) {
      fields.push(
        new Attribute({
          name: attribute.name,
          value: attribute.value,
        })
      )
    }
  }
  for (const predicate of data.resolvedPredicates) {
    fields.push(
      new Predicate({
        name: predicate.name,
        pType: predicate.predicateType,
        pValue: predicate.predicateValue,
      })
    )
  }
  return fields
}
