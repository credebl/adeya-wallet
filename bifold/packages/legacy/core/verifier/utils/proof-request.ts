import {
  Agent,
  AgentMessage,
  AttributeFilter,
  AttributeValue,
  AutoAcceptProof,
  ProofAttributeInfo,
  ProofExchangeRecord,
  ProofPredicateInfo,
  ProofRequest,
  V1RequestPresentationMessage,
} from '@aries-framework/core'

import { ProofRequestTemplate, ProofRequestType } from '../types/proof-reqeust-template'

const protocolVersion = 'v1'
const domain = 'http://aries-mobile-agent.com'

/*
 * Find Proof Request message in the storage by the given id
 * */
export const findProofRequestMessage = async (agent: Agent, id: string): Promise<ProofRequest | undefined> => {
  const message = await agent.proofs.findRequestMessage(id)
  if (message && message instanceof V1RequestPresentationMessage && message.indyProofRequest) {
    return message.indyProofRequest
  } else {
    return undefined
  }
}

const convertCase: { [key: string]: keyof AttributeFilter | undefined } = {
  schema_id: 'schemaId',
  schema_issuer_did: 'schemaIssuerDid',
  schema_name: 'schemaName',
  schema_version: 'schemaVersion',
  issuer_did: 'issuerDid',
  cred_def_id: 'credentialDefinitionId',
}

const fromAnonCreds = (restriction: Record<string, string>): AttributeFilter => {
  const restrictionObject = {} as AttributeFilter
  Object.keys(restriction).forEach((anonKey) => {
    if (anonKey.startsWith('attr::') && anonKey.endsWith('::value')) {
      restrictionObject.attributeValue = new AttributeValue({
        name: anonKey.split('::')[1],
        value: restriction[anonKey],
      })
    } else {
      const key = convertCase[anonKey]
      if (key) {
        restrictionObject[key] = restriction[anonKey] as any
      }
    }
  })
  return restrictionObject
}

/*
 * Build Proof Request data from for provided template
 * */
/*
 * Build Proof Request data for provided template
 * */
export const buildProofRequestDataForTemplate = (
  template: ProofRequestTemplate,
  customValues?: Record<string, Record<string, number>>
) => {
  if (template.payload.type === ProofRequestType.Indy) {
    const requestedAttributes: Map<string, ProofAttributeInfo> = new Map()
    const requestedPredicates: Map<string, ProofPredicateInfo> = new Map()
    let index = 0

    template.payload.data.forEach((data) => {
      if (data.requestedAttributes?.length) {
        data.requestedAttributes.forEach((requestedAttribute) => {
          const attribute = new ProofAttributeInfo({
            name: requestedAttribute.name,
            names: requestedAttribute.names,
            nonRevoked: requestedAttribute.nonRevoked,
            restrictions: requestedAttribute.restrictions?.map((restriction) => fromAnonCreds(restriction)),
          })
          requestedAttributes.set(`referent_${index}`, attribute)
          index++
        })
      }
      if (data.requestedPredicates?.length) {
        data.requestedPredicates.forEach((requestedPredicate) => {
          const customValue =
            customValues && customValues[data.schema] ? customValues[data.schema][requestedPredicate.name] : undefined

          const predicate = new ProofPredicateInfo({
            name: requestedPredicate.name,
            predicateValue:
              requestedPredicate.parameterizable && customValue ? customValue : requestedPredicate.predicateValue,
            predicateType: requestedPredicate.predicateType,
            nonRevoked: requestedPredicate.nonRevoked,
            restrictions: requestedPredicate.restrictions?.map((restriction) => fromAnonCreds(restriction)),
          })

          requestedPredicates.set(`referent_${index}`, predicate)
          index++
        })
      }
    })
    return {
      indy: {
        name: template.name,
        version: template.version,
        nonce: Date.now().toString(),
        requestedAttributes,
        requestedPredicates,
      },
    }
  }
  if (template.payload.type === ProofRequestType.DIF) {
    return {}
  }
}

export interface CreateProofRequestInvitationResult {
  request: AgentMessage
  proofRecord: ProofExchangeRecord
  invitation: AgentMessage
  invitationUrl: string
}

/*
 * Create connectionless proof request invitation for provided template
 * */
export const createConnectionlessProofRequestInvitation = async (
  agent: Agent,
  template: ProofRequestTemplate,
  customPredicateValues?: Record<string, Record<string, number>>
): Promise<CreateProofRequestInvitationResult | undefined> => {
  const proofFormats = buildProofRequestDataForTemplate(template, customPredicateValues)
  if (!proofFormats) {
    return undefined
  }
  const { message: request, proofRecord } = await agent.proofs.createRequest({
    protocolVersion,
    autoAcceptProof: AutoAcceptProof.Always,
    proofFormats,
  })
  const { message: invitation, invitationUrl } = await agent.oob.createLegacyConnectionlessInvitation({
    recordId: proofRecord.id,
    message: request,
    domain,
  })
  return {
    request,
    proofRecord,
    invitation,
    invitationUrl,
  }
}

export interface SendProofRequestResult {
  proofRecord: ProofExchangeRecord
}

/*
 * Build Proof Request for provided template and send it to provided connection
 * */
export const sendProofRequest = async (
  agent: Agent,
  template: ProofRequestTemplate,
  connectionId: string,
  customPredicateValues?: Record<string, Record<string, number>>
): Promise<SendProofRequestResult | undefined> => {
  const proofFormats = buildProofRequestDataForTemplate(template, customPredicateValues)
  if (!proofFormats) {
    return undefined
  }
  const proofRecord = await agent.proofs.requestProof({
    protocolVersion,
    connectionId,
    proofFormats,
  })
  return {
    proofRecord,
  }
}

/*
 * Check if the Proof Request template contains at least one predicate
 * */
export const hasPredicates = (record: ProofRequestTemplate): boolean => {
  if (record.payload.type === ProofRequestType.Indy) {
    return record.payload.data.some((d) => d.requestedPredicates && d.requestedPredicates?.length > 0)
  }
  if (record.payload.type === ProofRequestType.DIF) {
    return false
  }
  return false
}

/*
 * Check if the Proof Request template contains parameterizable predicates
 * */
export const isParameterizable = (record: ProofRequestTemplate): boolean => {
  if (record.payload.type === ProofRequestType.Indy) {
    return record.payload.data.some((d) => d.requestedPredicates?.some((predicate) => predicate.parameterizable))
  }
  if (record.payload.type === ProofRequestType.DIF) {
    return false
  }
  return false
}
