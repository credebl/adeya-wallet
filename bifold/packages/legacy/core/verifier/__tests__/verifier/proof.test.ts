import { getProofIdentifiers, groupSharedProofDataByCredential, parseIndyProof } from '../../utils/proof'
import { IndyProof } from 'indy-sdk-react-native'

const proof_request = {
  name: 'proof-request',
  version: '1.0',
  nonce: '1298236324864',
  requested_attributes: {
    attribute_1: {
      names: ['first_name', 'second_name'],
    },
    attribute_2: {
      name: 'date',
    },
    attribute_3: {
      name: 'passport_no',
    },
  },
  requested_predicates: {
    predicate_1: {
      name: 'age',
      p_type: '>=' as any,
      p_value: 1,
    },
  },
}

const proof: IndyProof = {
  proof: {
    proofs: [],
    aggregated_proof: {
      c_list: [],
    },
  },
  requested_proof: {
    revealed_attrs: {
      attribute_2: {
        sub_proof_index: 0,
        raw: '1/1/1',
        encoded: '',
      },
      attribute_3: {
        sub_proof_index: 1,
        raw: '123123',
        encoded: '',
      },
    },
    revealed_attr_groups: {
      attribute_1: {
        sub_proof_index: 0,
        values: {
          first_name: {
            raw: 'Aries',
            encoded: '',
          },
          second_name: {
            raw: 'Bifold',
            encoded: '',
          },
        },
      },
    },
    self_attested_attrs: {},
    unrevealed_attrs: {},
    // @ts-ignore
    predicates: {
      predicate_1: {
        sub_proof_index: 0,
      },
    },
  },
  identifiers: [
    {
      schema_id: '7KuDTpQh3GJ7Gp6kErpWvM:2:Faber College:1.0.0',
      cred_def_id: '7KuDTpQh3GJ7Gp6kErpWvM:3:CL:712167:latest',
    },
    {
      schema_id: '7KuDTpQh3GJ7Gp6kErpWvM:2:Passport:1.0.0',
      cred_def_id: '7KuDTpQh3GJ7Gp6kErpWvM:3:CL:712168:latest',
    },
  ],
}

describe('Helpers', () => {
  test('Parse indy proof', async () => {
    const proofRequest = parseIndyProof(proof_request, proof)
    expect(proofRequest).toMatchSnapshot()
  })

  test('Group shared proof data by credential', async () => {
    const data = parseIndyProof(proof_request, proof)
    const groupedData = groupSharedProofDataByCredential(data)
    expect(groupedData).toMatchSnapshot()
  })

  test('Get proof identifiers', async () => {
    const identifiers = getProofIdentifiers(proof, 0)
    expect(identifiers).toMatchSnapshot()
  })
})
