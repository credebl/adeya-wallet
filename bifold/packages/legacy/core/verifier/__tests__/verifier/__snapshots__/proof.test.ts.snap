// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Helpers Parse indy proof 1`] = `
ParsedIndyProof {
  "resolvedPredicates": Array [
    Object {
      "identifiers": Object {
        "credentialDefinitionId": "7KuDTpQh3GJ7Gp6kErpWvM:3:CL:712167:latest",
        "revocationRegistryId": undefined,
        "schemaId": "7KuDTpQh3GJ7Gp6kErpWvM:2:Faber College:1.0.0",
        "timestamp": undefined,
      },
      "name": "age",
      "predicateType": ">=",
      "predicateValue": 1,
    },
  ],
  "sharedAttributeGroups": Array [
    Object {
      "attributes": Array [
        Object {
          "name": "first_name",
          "value": "Aries",
        },
        Object {
          "name": "second_name",
          "value": "Bifold",
        },
      ],
      "identifiers": Object {
        "credentialDefinitionId": "7KuDTpQh3GJ7Gp6kErpWvM:3:CL:712167:latest",
        "revocationRegistryId": undefined,
        "schemaId": "7KuDTpQh3GJ7Gp6kErpWvM:2:Faber College:1.0.0",
        "timestamp": undefined,
      },
    },
  ],
  "sharedAttributes": Array [
    Object {
      "identifiers": Object {
        "credentialDefinitionId": "7KuDTpQh3GJ7Gp6kErpWvM:3:CL:712167:latest",
        "revocationRegistryId": undefined,
        "schemaId": "7KuDTpQh3GJ7Gp6kErpWvM:2:Faber College:1.0.0",
        "timestamp": undefined,
      },
      "name": "date",
      "value": "1/1/1",
    },
    Object {
      "identifiers": Object {
        "credentialDefinitionId": "7KuDTpQh3GJ7Gp6kErpWvM:3:CL:712168:latest",
        "revocationRegistryId": undefined,
        "schemaId": "7KuDTpQh3GJ7Gp6kErpWvM:2:Passport:1.0.0",
        "timestamp": undefined,
      },
      "name": "passport_no",
      "value": "123123",
    },
  ],
  "unresolvedAttributeGroups": Array [],
  "unresolvedAttributes": Array [],
  "unresolvedPredicates": Array [],
}
`;

exports[`Helpers Group shared proof data by credential 1`] = `
Map {
  "7KuDTpQh3GJ7Gp6kErpWvM:3:CL:712167:latest" => Object {
    "data": CredentialSharedProofData {
      "resolvedPredicates": Array [
        Object {
          "identifiers": Object {
            "credentialDefinitionId": "7KuDTpQh3GJ7Gp6kErpWvM:3:CL:712167:latest",
            "revocationRegistryId": undefined,
            "schemaId": "7KuDTpQh3GJ7Gp6kErpWvM:2:Faber College:1.0.0",
            "timestamp": undefined,
          },
          "name": "age",
          "predicateType": ">=",
          "predicateValue": 1,
        },
      ],
      "sharedAttributeGroups": Array [
        Object {
          "attributes": Array [
            Object {
              "name": "first_name",
              "value": "Aries",
            },
            Object {
              "name": "second_name",
              "value": "Bifold",
            },
          ],
          "identifiers": Object {
            "credentialDefinitionId": "7KuDTpQh3GJ7Gp6kErpWvM:3:CL:712167:latest",
            "revocationRegistryId": undefined,
            "schemaId": "7KuDTpQh3GJ7Gp6kErpWvM:2:Faber College:1.0.0",
            "timestamp": undefined,
          },
        },
      ],
      "sharedAttributes": Array [
        Object {
          "identifiers": Object {
            "credentialDefinitionId": "7KuDTpQh3GJ7Gp6kErpWvM:3:CL:712167:latest",
            "revocationRegistryId": undefined,
            "schemaId": "7KuDTpQh3GJ7Gp6kErpWvM:2:Faber College:1.0.0",
            "timestamp": undefined,
          },
          "name": "date",
          "value": "1/1/1",
        },
      ],
    },
    "identifiers": Object {
      "credentialDefinitionId": "7KuDTpQh3GJ7Gp6kErpWvM:3:CL:712167:latest",
      "revocationRegistryId": undefined,
      "schemaId": "7KuDTpQh3GJ7Gp6kErpWvM:2:Faber College:1.0.0",
      "timestamp": undefined,
    },
  },
  "7KuDTpQh3GJ7Gp6kErpWvM:3:CL:712168:latest" => Object {
    "data": CredentialSharedProofData {
      "resolvedPredicates": Array [],
      "sharedAttributeGroups": Array [],
      "sharedAttributes": Array [
        Object {
          "identifiers": Object {
            "credentialDefinitionId": "7KuDTpQh3GJ7Gp6kErpWvM:3:CL:712168:latest",
            "revocationRegistryId": undefined,
            "schemaId": "7KuDTpQh3GJ7Gp6kErpWvM:2:Passport:1.0.0",
            "timestamp": undefined,
          },
          "name": "passport_no",
          "value": "123123",
        },
      ],
    },
    "identifiers": Object {
      "credentialDefinitionId": "7KuDTpQh3GJ7Gp6kErpWvM:3:CL:712168:latest",
      "revocationRegistryId": undefined,
      "schemaId": "7KuDTpQh3GJ7Gp6kErpWvM:2:Passport:1.0.0",
      "timestamp": undefined,
    },
  },
}
`;

exports[`Helpers Get proof identifiers 1`] = `
Object {
  "credentialDefinitionId": "7KuDTpQh3GJ7Gp6kErpWvM:3:CL:712167:latest",
  "revocationRegistryId": undefined,
  "schemaId": "7KuDTpQh3GJ7Gp6kErpWvM:2:Faber College:1.0.0",
  "timestamp": undefined,
}
`;
